libs/application/template-api-modules/src/lib/modules/shared/api/national-registry/national-registry.service.ts

```diff
@@ -31,11 +31,34 @@ export class NationalRegistryService extends BaseTemplateApiService {
     auth,
     params,
   }: TemplateApiModuleActionProps<NationalRegistryParameters>): Promise<NationalRegistryIndividual | null> {
```

Comment:
This functionality complicates this function somewhat. So I would suggest breaking this up like so. 
``` 
const result = await this.getIndividual(auth.nationalId);

if (!result) {
        throw new TemplateApiError(
            {
                title: coreErrorMessages.nationalIdNotFoundInNationalRegistryTitle,
                summary: coreErrorMessages.nationalIdNotFoundInNationalRegistrySummary,
            },
            400
        );
    }
    
    this.validateIndividual(result, params);
    
    if (params?.includeChildrenInValidation) {
        const children = await this.nationalRegistryApi.getCustodyChildren(auth);
        await this.validateChildren(children, params);
    }

    return result;

``` 
Where validateChildren loops through each child uses the validateIndividual and  then a separate function for each validation.

Reply:
I've broken up the function, makes the validation part easier to read